lib/license_finder/decisions.rb

```diff
@@ -115,71 +118,104 @@ def approve(name, txn = {})
     end
 
     def unapprove(name, txn = {})
-      @decisions << [:unapprove, name, txn]
+      add_decision [:unapprove, name, txn]
       @approvals.delete(name)
       self
     end
 
     def permit(lic, txn = {})
-      @decisions << [:permit, lic, txn]
+      add_decision [:permit, lic, txn]
       @permitted << License.find_by_name(lic)
       self
     end
 
     def unpermit(lic, txn = {})
-      @decisions << [:unpermit, lic, txn]
+      add_decision [:unpermit, lic, txn]
       @permitted.delete(License.find_by_name(lic))
       self
     end
 
     def restrict(lic, txn = {})
-      @decisions << [:restrict, lic, txn]
+      add_decision [:restrict, lic, txn]
       @restricted << License.find_by_name(lic)
       self
     end
 
     def unrestrict(lic, txn = {})
-      @decisions << [:unrestrict, lic, txn]
+      add_decision [:unrestrict, lic, txn]
       @restricted.delete(License.find_by_name(lic))
       self
     end
 
     def ignore(name, txn = {})
-      @decisions << [:ignore, name, txn]
+      add_decision [:ignore, name, txn]
       @ignored << name
       self
     end
 
     def heed(name, txn = {})
-      @decisions << [:heed, name, txn]
+      add_decision [:heed, name, txn]
       @ignored.delete(name)
       self
     end
 
     def ignore_group(name, txn = {})
-      @decisions << [:ignore_group, name, txn]
+      add_decision [:ignore_group, name, txn]
       @ignored_groups << name
       self
     end
 
     def heed_group(name, txn = {})
-      @decisions << [:heed_group, name, txn]
+      add_decision [:heed_group, name, txn]
       @ignored_groups.delete(name)
       self
     end
 
     def name_project(name, txn = {})
-      @decisions << [:name_project, name, txn]
+      add_decision [:name_project, name, txn]
       @project_name = name
       self
     end
 
     def unname_project(txn = {})
-      @decisions << [:unname_project, txn]
+      add_decision [:unname_project, txn]
       @project_name = nil
       self
     end
 
+    def inherit_from(filepath)
+      decisions =
+        if filepath =~ %r{^https?://}
+          # ruby < 2.5.0 URI.open is private
+          # rubocop:disable Security/Open
+          open(filepath).read
+          # rubocop:enable Security/Open
```

Comment:
I already used `open` in the `package_url` PR but rubocop did not complained about it. Here it does but for ruby 2.3.8 and ruby 2.4.9 it fails by using `URI.open` since for those versions this method is private. What do you suggest? Disable rubocop here or having a if/else block where I check the ruby version?

Reply:
Hmm....good catch. I wonder if it was fine before because if the nested nature of how it was used so rubocop doesn't notice it. If URI.open is not available on 2.3 or 2.4, we may need to disable rubocop here for those versions as you have done. I would add the conditional for the versions though. Rubocop in the pipeline does currently run on the 2.7.1 ruby version of the ruby docker image so there shouldn't be an issue.

Reply:
I have added the version conditional as requested but kept the rubocop disable for the used `open` call for ruby < 2.5.0 otherwise rubocop would fail on development systems which I think should not happen.